src/core/namespace.c

```diff
@@ -1756,6 +1790,10 @@ int setup_namespace(
                 goto finish;
         }
 
+        /* Create the source directory to allow runtime propagation of mounts */
+        if (setup_propagate)
+                (void) mkdir_p(strjoina(propagate_dir, "/", unit_id), 0600);
```

Comment:
This one is not ok.

Reply:
reworked slightly, no point in passing unit_id here, we just care about the propagate directory